providers/amazon/src/airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -58,12 +58,16 @@
 from airflow.providers.amazon.aws.utils.connection_wrapper import AwsConnectionWrapper
 from airflow.providers.amazon.aws.utils.identifiers import generate_uuid
 from airflow.providers.amazon.aws.utils.suppress import return_on_error
+from airflow.providers.common.compat.version_compat import AIRFLOW_V_3_0_PLUS
 from airflow.providers_manager import ProvidersManager
 from airflow.utils.helpers import exactly_one
 from airflow.utils.log.logging_mixin import LoggingMixin
 
 BaseAwsConnection = TypeVar("BaseAwsConnection", bound=Union[boto3.client, boto3.resource])
 
+if AIRFLOW_V_3_0_PLUS:
+    from airflow.sdk.exceptions import AirflowRuntimeError
```

Comment:
What if Airflow < 3, `AirflowRuntimeError` is undefined?

Reply:
Yeah another branch is needed below to optionally use this Exception.

Reply:
Let me know how you feel about the way I did it